clients/drcachesim/tracer/tracer.cpp

```diff
@@ -540,6 +540,8 @@ append_marker_seg_base(void *drcontext, trace_marker_type_t marker, uintptr_t va
         return; /* This thread was filtered out. */
     BUF_PTR(data->seg_base) +=
         instru->append_marker(BUF_PTR(data->seg_base), marker, value);
+    if (file_ops_func.handoff_buf == NULL)
+        memtrace(drcontext, false);
```

Comment:
I don't think we want to call this every time: we're going to end up dumping a single-entry buffer over and over for the retaddr and func arg markers.  We need to add a check for whether the buffer has hit the redzone (in C++ here -- today we only have the check in the generated instru).

Reply:
Done.